### PR TITLE
Feat

### PR DESCRIPTION
**Pull request by KumarAmitt**
**Project - Enumerable Methods**

-- Implement custom enumerable methods

`my_each`
`my_each_with_index`
`my_select`
`my_all?`
`my_any?`
`my_none?`
`my_count`
`my_map`
`my_inject`
`multiply_els`

-- Any feedback and suggestion to improve this project will be highly appreciated.

### CODE REVIEW

**Review by andresporras3423 — CHANGES_REQUESTED** [review]
### Status: Changes Required :recycle:
**Hello Ralph Oburu & Amit Kumar** 
Good job with your first submission for this project :muscle:  :muscle:
It passes most of the tests :clap: :clap: :clap:  
There are just a few details to improve before you can go ahead:

- [ ] Fix ruboop offenses
- [ ] Add a screenshot in the README file
- [ ] #my_each_with_index when a block is given when self is an array
     ```
      concat1=''
      [1,2,3,4].each_with_index(&proc{|n, i| concat1+='n: '+n.to_s+', i: '+i.to_s+'+++'})
      concat2=''
      [1,2,3,4].each_with_index(&proc{|n, i| concat2+='n: '+n.to_s+', i: '+i.to_s+'+++'})
      concat1 == concat2
     ```
       behaves like #each_with_index calls the given block once for each element in self
- [ ] #my_each_with_index when a block is given when self is a range
     ```
      concat1=''
      (1..4).each_with_index(&proc{|n, i| concat1+='n: '+n.to_s+', i: '+i.to_s+'+++'})
      concat2=''
      (1..4).each_with_index(&proc{|n, i| concat2+='n: '+n.to_s+', i: '+i.to_s+'+++'})
      concat1 == concat2
     ```
       behaves like #each_with_index calls the given block once for each element in self
- [ ] #my_each_with_index when a block is given when self is a hash
     ```
      concat1=''
      { a: 1, b: 2, c: 3, d: 4}.each_with_index(&proc{|n, i| concat1+='n: '+n.to_s+', i: '+i.to_s+'+++'})
      concat2=''
      { a: 1, b: 2, c: 3, d: 4}.each_with_index(&proc{|n, i| concat2+='n: '+n.to_s+', i: '+i.to_s+'+++'})
      concat1 == concat2
    ```
       behaves like #each_with_index calls the given block once for each element in self
 
Please request another review once you are ready! :+1: 
 
Happy coding :relaxed:
 
Oscar Andrés Russi Porras

File: enumerable.rb [hunk 1]
```diff
@@ -0,0 +1,142 @@
...
+    arr.length.times do |i|
+      yield to_a[i]
+    end
+
+    self
+  end
+
+  # 2. my_each_with_index Method-----------------------------
+
+  def my_each_with_index
```
Comment (mcrd25) [inline enumerable.rb:18]: You technically give the right answer but you are giving the wrong output. It's an array

Comment (mcrd25) [reply enumerable.rb:18]: Screenshot from 2021-01-08 11-47-04

Comment (mcrd25) [reply enumerable.rb:18]: Screenshot from 2021-01-08 11-47-47

**Review by mcrd25 — CHANGES_REQUESTED** [review]
# Changes Required

Hi Team :wave: 

Good job so far. There only 3 tests failing on my end.

### Required Changes
- Fix issues with `my_each_with_index`

***Please makes changes and submit another review request.
If you have any questions, let me know here or email (preferably) or on slack and I will be glad to clarify with you. You can also reach me and connect with me on my social media or email.***

If you liked my review, consider upskilling me on LinkedIn. And please do connect. [My LinkedIn](https://www.linkedin.com/in/mayadouglas/)

*Happy coding 😄,*
**[--Maya Douglas](mayacdouglas.com)**
[GitHub](https://github.com/mcrd25) [Twitter](https://twitter.com/mcrd25) [linkedin](https://www.linkedin.com/in/mayadouglas/) [GitLab](https://gitlab.com/mcrd25)

---
*P.S. all requested changes not marked by [OPTIONAL] or are not positive comments are **REQUIRED***

**Review by ebeagusamuel — CHANGES_REQUESTED** [review]
# Changes Required :recycle: 
### Hey guys
#### You are almost there. There are just two test cases failing. 
- [ ] enumerables #my_each when a block is given when self is a hash behaves like #each calls the given block once for each element in self.
- [ ] enumerables #my_each_with_index when a block is given when self is a hash behaves like #each_with_index calls the given block once for each element in self.
### Once these are fixed are done, then your project can be approved. Do request another review when you make the fixes.
slack(@ebeagusamuel)

**Review by javitocor — APPROVED** [review]
# STATUS APPROVED 🥇 

Congratulations team! 🥳 
You have made a great job with the enumerables 👍 
Remember to merge it, fill up the completion form 📑  and you can move into the next project 🚀 
Good luck with it! 😃

**Pull request merged**